homeassistant/components/screenlogic/entity.py

```diff
@@ -109,7 +109,7 @@ def _async_data_updated(self) -> None:
 
```

Comment:
We should probably not write state if it hasn't changed.

Reply:
I worried about that, but it looked like `async_set` would catch and prevent that. https://github.com/home-assistant/core/blob/dbb2706c76de3c04958c3d0d97cd052106805767/homeassistant/core.py#L1536-L1541

Reply:
`async_set` is a bit late in the cycle since it has call the `state` generation function and to call all the properties and build the `attributes` every time to compare them, decide they are the same, and than do nothing.

Reply:
Ideally the coordinator callback would only write state when availability changes if its a push entity